content/blog/2016/2016-05-26-gsoc2016-automatic-plugin-documentation.adoc

```diff
@@ -0,0 +1,31 @@
+---
+layout: post
+title: "Automatic Plugin Documentation"
+tags:
+- jenkins
+- gsoc
+- plugins
+author: cynthia
+---
+=== About me 
+I am Cynthia Anyango from Nairobi , Kenya . I am a second yeat student at Maseno University . I am currently specializing on Ruby on Rails and trying to learn Python . I recently started contributing to Open source projects . My major contribution was at Mozilla , where I worked with the QA for Cloud services . I did manual and automated tests for various cloud services  . I wrote documentation too . Above that , I am competent and I am always passionate about what I get my hands on . 
+ 	
+=== Project summary 
+
+Currently Jenkins has plugin documentation is being stored in Confluence. Sometimes the documentation is scattered and outdated. In order to improve the situation we would like to follow the documentation-as-code approach and to put docs to plugin repositories and then publish them on the project website using the awestruct engine. The project aims an implementation of a documentation continuous deployment flow powered by Jenkins and Pipeline Plugin.
+
+The idea is to automatically pull in the README and other docs from GitHub, show changelogs with versions and releases dates . I will be designing file templates that will contain most of the  docs information that will be required from plugin developers . The file templates will be written in ASCIIDOC .  Plugin developers will get a chance to review the templates. The templates will be prototyped by various plugin developers . 
```

Comment:
> "that will contain most of the docs information that will be required from plugin developers

"docs information" makes this confusing to me, I'm not sure I understand the statement being made here

----


The proper casing is "AsciiDoc", not "ASCIIDOC"

----

I think it's fine for us to be a bit opinionated at the start for chosing AsciiDoc, but I don't think that will be the *only* file format we *ever* accept. So I suggest rephrasing "_The file templates will be written in AsciiDoc_" to "Initially, the files will be written in AsciiDoc_"

I also suggest linking "AsciiDoc" to asciidoctor.org